### PR TITLE
feat: add function to create model from dict

### PR DESCRIPTION
**Pull request by h4shk4t**
Goals:

This PR is for issue #1191 

With this PR a user will be able to directly create models from a dictionary without explicitly specifying the types of the values. The function will automatically "guess" its type and generate a TypedDict class from the "guessed" types.

### CODE REVIEW

**Comment by samsja** [comment]
Hey @h4shk4t . Thanks for your contribution. Can you please add a more detailed description of what you are doing in your PR ?

And add test as well our the different functionality you are adding :)

File: docarray/documents/helper.py [hunk 1]
```diff
@@ -118,3 +118,24 @@ class MyAudio(TypedDict):
     doc = create_model_from_typeddict(typeddict_cls, **kwargs)
 
     return doc
+
+def create_from_dict(
+    data: Dict[str,Any],  # type: ignore
+    **kwargs: Any,
+):
```
Comment (samsja) [inline docarray/documents/helper.py:125]: Don't forget to add docstring here

File: docarray/documents/helper.py [hunk 2]
```diff
@@ -118,3 +118,24 @@ class MyAudio(TypedDict):
     doc = create_model_from_typeddict(typeddict_cls, **kwargs)
 
     return doc
+
+def create_from_dict(
+    data: Dict[str,Any],  # type: ignore
```
Comment (samsja) [inline docarray/documents/helper.py:123]: why type ignore here ?

Comment (h4shk4t) [reply docarray/documents/helper.py:123]: I did not add any type check here as I thought the dictionary could contain any type as values so I used `Any` here. Should I do something else here?

File: docarray/documents/helper.py [hunk 3]
```diff
@@ -118,3 +118,24 @@ class MyAudio(TypedDict):
...
+        if not issubclass(kwargs['__base__'], BaseDocument):
+            raise ValueError(
+                f'{kwargs["__base__"]} is not a BaseDocument or its subclass'
+            )
+    else:
+        kwargs['__base__'] = BaseDocument
+    typed_dict_fields = {}
+    for key, value in data.items():
+        field_type = type(value)
+        typed_dict_fields[key] = field_type
```
Comment (jupyterjazz) [inline docarray/documents/helper.py:136]:
```suggestion
    typed_dict_fields: Dict[str, Any] = {key: type(val) for key, val in data.items()}
```

File: docarray/documents/helper.py [hunk 4]
```diff
@@ -118,3 +118,24 @@ class MyAudio(TypedDict):
...
+        print(typed_dict_fields[key])
```
Comment (jupyterjazz) [inline docarray/documents/helper.py:137]:
```suggestion
```

**Review by jupyterjazz — COMMENTED** [review]
thanks for contributing @h4shk4t!
please don't forget to add tests

**Comment by jupyterjazz** [comment]
There's another PR for this issue so I'm gonna close this one.
Thank you for your interest! and feel free to tackle another ticket 💪

**Pull request closed without merging**